src/App.tsx

```diff
@@ -1,14 +1,61 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { Status } from './types/Status';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [selectedStatus, setSelectedStatus] = useState(Status.All);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        throw new Error('Please try again later');
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  function filterTodos(
+    items: Todo[],
+    status: Status,
+    queryValue: string,
+  ) {
+    let filtered = [...items];
```

Comment:
I suggest you to using two variables, it will make you code cleaner 
 - filteredByName
 - filteredByStatus

Reply:
thank you for the suggestion! but as for me now it looks more complicated